draft-ietf-tls-esni.md

```diff
@@ -952,6 +978,20 @@ Additionally, this specification allows for clients to send GREASE ESNI
 extensions (see {{grease-extensions}}), which helps ensure the ecosystem
 handles the values correctly.
 
+The additional construction in {{record-digest-mac}} is intended to prevent an
+attacker from being able to distinguish between a real and GREASE ESNI
+extension. An attacker that is able to distinguish the two cases could block
+only real ESNI connections. This may disincentivize servers from deploying ESNI
+as they'd suffer additional breakage when they enable ESNI. When real and GREASE
+ESNI extensions are indistinguishable, an attacker wouldn't be able to block
+connections based on whether the server supports ESNI.
+
+This construction could be defeated by an attacker which has a list of ESNIKeys
+structures it wishes to block, as the attacker could just MAC each possibility
+and compare against what the client sent. However, this is roughly equivalent to
+the attacker blocking the connection based on the destination IP address.
```

Comment:
I'm not sure how persuasive I find this. It seems like it's quite likely that the attacker can determine which IP addresses support ESNI, simply by doing reverse resolution and then forward resolution. Why does this help?

Reply:
A way of avoiding that, though which might be problematic to deploy, is that the reverse resolution name and the public name don't have ESNI records, since you don't necessarily need ESNI when hitting those names, since they're already visible to an attacker.